backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/CombinedWalletClientImpl.kt

```diff
@@ -135,20 +157,42 @@ class CombinedWalletClientImpl private constructor(
     private suspend fun remove(serviceMode: ServiceMode) =
         withContext(Dispatchers.Default) { cache.remove(serviceMode) }
 
-    private suspend fun getOrCreate(serviceMode: ServiceMode) =
-        withContext(Dispatchers.Default) { cache.getOrPut(serviceMode) { create() } }
+    private suspend fun getOrCreate(serviceMode: ServiceMode): PartialWalletClient =
+        withContext(Dispatchers.Default) {
+            val partialClient = cache[serviceMode]
+            if (partialClient == null) {
+                val newClient = create()
+                cache[serviceMode] = newClient
+                newClient
+            } else {
+                partialClient
+            }
```

Comment:
This looks like it can be unaltered now, back to `cache.getOrPut(serviceMode) { create() }`.